packages/profiles/source/transaction-index.ts

```diff
@@ -59,11 +59,11 @@ export class TransactionIndex implements ITransactionIndex {
 	async #fetch(query: Services.ClientTransactionsInput): Promise<ExtendedConfirmedTransactionDataCollection> {
 		const result = await this.#wallet.getAttributes().get<Coins.Coin>("coin").client().transactions(query);
 
-		for (const transaction of result.items()) {
-			if (this.#wallet.isCold() && (transaction.isSent() || transaction.isReturn())) {
-				this.#wallet.data().set(WalletData.Status, WalletFlag.Hot);
-			}
+		if (result.items().length > 0 && this.#wallet.isCold()) {
+			this.#wallet.data().set(WalletData.Status, WalletFlag.Hot);
```

Comment:
a cold wallet is a wallet that has not sent anything. Once the wallet itself has an outgoing transaction it should be considered hot, so it's not related to any incoming transactions to it